### PR TITLE
Allow setting 'bucket.storage_class'.

### LINKED ISSUE #1030: How do I create a Nearline / DRA bucket in Cloud Storage?
Page Name: storage-buckets
Release: 0.7.0

http://googlecloudplatform.github.io/gcloud-python/latest/storage-buckets.html#gcloud.storage.bucket.Bucket.create doesn't make this all that clear.

### PR DESCRIPTION
**Pull request by tseaver**
Fixes #1030.

### CODE REVIEW

File: gcloud/storage/test_bucket.py [hunk 1]
```diff
@@ -569,6 +569,7 @@ def test_location_setter(self):
         self.assertEqual(bucket.location, None)
         bucket.location = 'AS'
         self.assertEqual(bucket.location, 'AS')
+        self.assertTrue('location' in bucket._changes)
```
Comment (dhermes) [inline gcloud/storage/test_bucket.py]: Thanks for adding this and its cousins below

**Comment by dhermes** [comment]
Only one question, why not add to the constructor? Or make the constructor take `**kwargs**` that we can farm out to setters.

**Comment by tseaver** [comment]
> why not add to the constructor?

We have four settable properties:  `cors`, `lifecycle_rules`, `storage_class`, and `versioning_enabled`:  we would likely need to allow any of them to be set, if we wanted this feature.

> Or make the constructor take **kwargs** that we can farm out to setters.

Ugh:  we lose Python's type checking, and have to add it in ourselves.

**Comment by dhermes** [comment]
How you mean type checking? Do the setters do any? Wouldn't they still?

**Comment by tseaver** [comment]
> How you mean type checking? Do the setters do any? Wouldn't they still?

That was an objection to using `**kw`, vs. explicitly-named arguments.  All of those properties are rare enough cases that we can leave them out of the constructor.

**Comment by dhermes** [comment]
Gotcha. So WDYT about just adding `storage_class` to the constructor (or all four)?

**Comment by tseaver** [comment]
>  WDYT about just adding storage_class to the constructor (or all four)?

:-1:  Seems like a complication for a pretty rare case, with an easy-to-understand alternative already present.

**Comment by dhermes** [comment]
OK LGTM

(Responding from email for first time ever. Like whoa. Weird.)

Danny Hermes

**Pull request merged**